### DISCUSSION TITLE
Create Web Component

### DISCUSSION THREAD

**Discussion by JuanICasareski** [quasarframework/quasar#13974] [General - Components / Directives / etc]
Is there a way to create a way using Quasar as a UI library?

**Comment by davidbludlow** [comment]
This can be done via the [Quasar UMD Flavor](https://quasar.dev/start/pick-quasar-flavour). But that solution is unacceptable for my use case because it doesn't allow tree shaking.

**Comment by yusufkandemir** [comment]
Quasar Vite plugin + Vite using the library mode is a good way to achieve this:
- https://quasar.dev/start/vite-plugin
- https://vuejs.org/guide/extras/web-components.html#building-custom-elements-with-vue
- https://vitejs.dev/guide/build.html#library-mode

You can output as UMD, ESM, or CJS depending on how you are aiming to distribute and consume your web component.

Here is an incomplete example:
https://stackblitz.com/edit/vitejs-vite-cpycim?file=src/main.ts,vite.config.ts,index.html,src/define-custom-element.ts
